extensions/amp-fresh/0.1/amp-fresh.js

```diff
@@ -26,14 +29,55 @@ export class AmpFresh extends AMP.BaseElement {
 
   /** @override */
   isLayoutSupported(layout) {
-    return isLayoutSizeDefined(layout);
+    return true;
   }
 
   buildCallback() {
     /** @private @const {boolean} */
     this.isExperimentOn_ = isExperimentOn(this.win, TAG);
 
     user().assert(this.isExperimentOn_, `Experiment ${TAG} disabled`);
+
+    /** @private @const {string} */
+    this.ampFreshId_ = user().assert(this.element.getAttribute('id'),
+        'amp-fresh must have an id.');
+
+    /** @private @const {!AmpFreshManager} */
+    this.manager_ = getOrInsallAmpFreshManager(this.element);
+
+    this.manager_.register(this.ampFreshId_, this);
+
+    this.hide();
+  }
+
+  /**
+   * @param {!Element} surrogateAmpFresh
+   */
+  update(surrogateAmpFresh) {
+    // Never reparent the surrogate to the current document's subtree
+    // as this will trigger custom element life cycles,
+    // importing it shouldn't trigger.
+    surrogateAmpFresh = this.win.document.adoptNode(surrogateAmpFresh);
+    this.mutateElement(() => {
+      this.element.textContent = '';
+      copyChildren(surrogateAmpFresh, this.element);
+      this.show();
+    });
+  }
+
+  /**
+   * Toggles the element to be visible and active.
+   */
+  show() {
```

Comment:
Might be worth renaming to better explain this method. E.g. `ready()` or `markFresh()`, etc.